pkg/apis/provisioning/v1alpha3/provisioner_defaults.go

```diff
@@ -39,48 +39,48 @@ func (s *ProvisionerSpec) SetDefaults(ctx context.Context) {}
 // The returned copy might be complemented by dynamic default values which
 // must not be hoisted (saved) into the original Provisioner CRD as those
 // default values might change over time (e.g. rolling upgrade of CABundle, ...).
-func (p *Provisioner) WithDynamicDefaults() (_ Provisioner, err error) {
+func (p *Provisioner) WithDynamicDefaults(ctx context.Context) (_ Provisioner, err error) {
 	provisioner := *p.DeepCopy()
-	provisioner.Spec, err = provisioner.Spec.withDynamicDefaults()
+	provisioner.Spec, err = provisioner.Spec.withDynamicDefaults(ctx)
 	return provisioner, err
 }
 
 // WithDefaults returns a copy of this ProvisionerSpec with some empty
 // properties replaced by default values.
-func (s *ProvisionerSpec) withDynamicDefaults() (_ ProvisionerSpec, err error) {
+func (s *ProvisionerSpec) withDynamicDefaults(ctx context.Context) (_ ProvisionerSpec, err error) {
 	spec := *s.DeepCopy()
-	spec.Cluster, err = spec.Cluster.withDynamicDefaults()
+	spec.Cluster, err = spec.Cluster.withDynamicDefaults(ctx)
 	return spec, err
 }
 
 // WithDefaults returns a copy of this Cluster with some empty
 // properties replaced by default values. Notably, it will try
 // to load the CABundle from the in-cluster configuraiton if it
 // is not explicitly set.
-func (c *Cluster) withDynamicDefaults() (_ Cluster, err error) {
+func (c *Cluster) withDynamicDefaults(ctx context.Context) (_ Cluster, err error) {
 	cluster := *c.DeepCopy()
-	cluster.CABundle, err = cluster.getCABundle()
+	cluster.CABundle, err = cluster.getCABundle(ctx)
 	return cluster, err
 }
 
-func (c *Cluster) getCABundle() (*string, error) {
+func (c *Cluster) getCABundle(ctx context.Context) (*string, error) {
```

Comment:
WDYT about removing the debug lines in this function? Regardless of where we get the CABundle from, this usually floods the logs with repetitive messages, even in the Debug mode. Is it possible to only log once from this? Maybe just not log at all?

Reply:
I want to cache these guys. I'll remove the log lines now and deal with the cache in another PR.